spec/list_spec.rb

```
@@ -0,0 +1,119 @@
+  end
+
+  it "responds to the method #show_menu" do
+    list = List.new
+    expect(list).to respond_to(:show_menu)
+  end
+
+  it "shows the menu" do
+    list = List.new
+    expect(list.show_menu).to eq({ "chinese" => 10, "pierogi" => 8, "curry" => 9, "carbonara" => 11, "salad" => 6 })
+  end
+
+  #receipt
+
+  it "contains an receipt of ordered dishes"do
+    list = List.new
+    expect(list.receipt).to eq({})
+  end
+
+  it "responds to method #show_receipt" do
+    list = List.new
+    expect(list).to respond_to(:show_receipt)
+  end
+
+  it "shows receipt" do
+    list = List.new
+    expect(list.show_receipt).to eq(list.receipt)
+  end
+
+
+  #add_dish
+
+  it "responds to the method #add_dish" do
+    list = List.new
+    expect(list).to respond_to(:add_dish).with(2).arguments
+  end
+
+  #it "adds a new dish to the receipt" do #I keep this test as temporarly pending for future learning purposes. It is not to be run in the code at any stage.
+    #list = List.new
+    # dish_name = "pierogi"
+    # quantity = 2
+    # expect(list.add_dish(dish_name, quantity)).to eq(list.receipt)
+    #
+    # dish_name = "pierogi2" #here I check what would happen if two dishes were added
+    # quantity = 4
+    # expect(list.add_dish(dish_name, quantity)).to eq(list.receipt)
+    #it is ok, because the receipt also changes after the first dish was added so the test passes
+  # end
+
+  it "adds a new dish to the receipt and prints the summary of last order "do
+    list = List.new
+    dish_name = "pierogi"
+    quantity = 2
+    expect(list.add_dish(dish_name, quantity)).to eq("You just have added to your order #{quantity} portions of #{dish_name}(s)")
+  end
+
+  it " tries to add a dish that is not in the menu "do
+    list = List.new
+    dish_name = "kotlety"
+    quantity = 2
+    expect(list.add_dish(dish_name, quantity)).to eq("This dish is not in the menu")
```

Should this throw an error instead of returning text?